pkg/config/log.go

```diff
@@ -25,26 +23,39 @@ import (
 const logFileMaxSize = 10 * 1024 * 1024         // 10MB
 const logDateFormat = "2006-01-02 15:04:05 MST" // see time.Format for format syntax
 
-var logCertPool *x509.CertPool
+var syslogTLSConfig *tls.Config
+
+func getSyslogTLSKeyPair() (*tls.Certificate, error) {
+	var syslogTLSKeyPair *tls.Certificate
+	if Datadog.IsSet("syslog_pem") && Datadog.IsSet("syslog_key") {
+		keypair, err := tls.LoadX509KeyPair(
+			Datadog.GetString("syslog_pem"),
+			Datadog.GetString("syslog_key"),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		syslogTLSKeyPair = &keypair
+	}
+
+	return syslogTLSKeyPair, nil
+}
 
 // SetupLogger sets up the default logger
-func SetupLogger(logLevel, logFile, uri string, rfc, tls bool, pem string, logToConsole bool, jsonFormat bool) error {
+func SetupLogger(logLevel, logFile, uri string, rfc, useTLS, logToConsole, jsonFormat bool) error {
 	var syslog bool
 
 	if uri != "" { // non-blank uri enables syslog
 		syslog = true
 	}
 
-	if pem != "" {
-		if cert, err := ioutil.ReadFile(pem); err == nil {
-			if logCertPool == nil {
-				logCertPool = x509.NewCertPool()
-			}
-			logCertPool.AppendCertsFromPEM(cert)
-		} else {
-			log.Errorf("Unable to read PEM certificate")
+	if useTLS {
+		syslogTLSKeyPair, err := getSyslogTLSKeyPair()
+		if err != nil {
 			return err
 		}
+		syslogTLSConfig = &tls.Config{Certificates: []tls.Certificate{*syslogTLSKeyPair}, InsecureSkipVerify: true}
```

Comment:
Do we want to force `InsecureSkipVerify` ?

Reply:
I actually think this should be an option, wanted to discuss it. What do you think?

Reply:
I think we should make this a option in any case.